low-level/sensors/impl/generic/node_sas_port.py

```diff
@@ -181,47 +185,135 @@ def initialize(self, conf_reader, msgQlist, product):
 
         return True
 
-    def check_and_send_alert(self, new_phy_link_count):
+    def sort_phy_list(self, phy_list):
+        """Sorts phy list based on phy num
+        """
+        # List to be returned
+        sorted_list = []
+        # Phy num to phy name mapping
+        phy_num_dict = {}
+        # List of phy nums
+        num_list = []
+        for phy in phy_list:
+            # Read phy num eg phy-0:12 -> 12
+            num = int(phy.split(':')[1])
+            num_list.append(num)
+            phy_num_dict[num] = phy
+        # Sort the numbers
+        num_list.sort()
+        for num in num_list:
+            sorted_list.append(phy_num_dict[num])
+        return sorted_list
+
+    def update_sas_ports_status(self):
+        """Reads current phy status and updates port connectivity status
+        """
+        # Assumption : phys will be present in multiples of 4
+        phy_list = [*self.phy_dir_to_linkrate_mapping]
+        phy_list = self.sort_phy_list(phy_list)
+
+        # Now we have a sorted list of phys
+        # Phys 0-3 for the 0th sas port, and so on in groups of 4 phys
+        # List containing status of all phys
+        hba = []
+        for phy in phy_list:
+            if self.phy_dir_to_linkrate_mapping[phy][1] == 'up':
+                hba.append(1)
+            else:
+                hba.append(0)
+
+        for i in range(0,self.NUM_SAS_PORTS):
+            # Save phy names forming this port for future use
+            self.port_phy_list_dict[i] = phy_list[4*i:4*i+4]
```

Comment:
The hardcoded 4 could have been constant the way we have the number of ports, could improve later.